internal/proxy/connection/manager.go

```diff
@@ -52,19 +52,49 @@ func (s *connectionManager) checkLoop() {
 			return
 		case <-t.C:
 			s.removeLongInactiveClients()
+			// not sure if we should purge them periodically.
+			s.purgeIfNumOfClientsExceed()
 			t.Reset(paramtable.Get().ProxyCfg.ConnectionCheckIntervalSeconds.GetAsDuration(time.Second))
 		}
 	}
 }
 
+func (s *connectionManager) purgeIfNumOfClientsExceed() {
+	if int64(s.clientInfos.Len()) < paramtable.Get().ProxyCfg.MaxConnectionNum.GetAsInt64() {
+		return
+	}
+
+	log.Info("number of client infos exceed limit, purge the oldest",
```

Comment:
how about move to bottom and also print the Duration of this purge?

Reply:
ok